src/rez/tests/test_shells.py

```diff
@@ -294,6 +372,25 @@ def _rex_appending():
 
         _execute_code(_rex_appending, expected_output)
 
+    @shell_dependent()
+    def test_variable_encoding(self):
```

Comment:
I think we could do without this test actually - these tests are really only asserting internal implementation details of the shells. If you go down this road (eg, testing that powershell class acts like powershell) you'd have to write loads more tests, and I don't really see the value in that. I would argue that that isn't what this test suite is for - it's for checking that rex code (which is shell agnostic by design) gives the expected result in all shells - no more.

Reply:
You are right.